src/anonymous-credentials/schema.ts

```diff
@@ -587,56 +624,78 @@ export class CredentialSchema extends Versioned {
     return `dk:${term}`;
   }
 
-  static flattenJSONSchema(node: any) {
-    if (typeof node.type !== 'string') {
-      throw new Error(`Schema node must have type field that is a string, got ${JSON.stringify(node)}`);
-    }
+  /**
+   * Convert a schema object as per JSON-schema syntax (`IJsonSchema`) to the internal representation (`ISchema`).
+   * Currently, does not check if the needed JSON-schema definitions are actually present but assumes that they will be
+   * already passed.
+   * @param node
+   * @param nodeKeyName - Name of the node, used for throwing more informative error message
+   */
+  static convertToInternalSchemaObj(node: any, nodeKeyName: string = ''): object {
+    // util function needed only in this func
+    const createFullName = (old: string, neww: string): string => {
+      return old.length == 0 ? neww : `${old}.${neww}`;
+    };
 
-    if (node.type === 'object') {
-      if (typeof node.properties !== 'undefined') {
-        const result: object = {};
-        const keys = Object.keys(node.properties);
-        keys.forEach((k) => {
-          const value = node.properties[k];
-
-          let insVal = {};
-          if (value.type === 'object') {
-            if (typeof value.properties === 'object') {
-              insVal = CredentialSchema.flattenJSONSchema(value);
-            } else {
-              throw new Error(`${k} must have properties field that is an object`);
-            }
-          } else if (value.type === 'array') {
-            if (Array.isArray(value.items)) {
-              insVal = value.items.map((i) => CredentialSchema.flattenJSONSchema(i));
-            } else {
-              throw new Error('No indefinite length array support');
-            }
+    // Will either have a "type" property or will be defined using "$ref"
+    const typ = node.type;
+    const ref = node.$ref;
+    if (typ !== undefined) {
+      switch (typ) {
+        case 'string':
+          return node;
+        case 'integer':
+          // If key "minimum" is not present or not an integer, validation logic in other function will throw.
+          return { type: this.INT_TYPE, minimum: node.minimum };
```

Comment:
again wrong as my internal type should be positive integer if min is greater or equal to 0

Reply:
Explained above.

Reply:
we discussed on the call as to why its beneficial to detect this way, can resolve this thread